src/unfold.js

```diff
@@ -3,35 +3,37 @@ var _curry2 = require('./internal/_curry2');
 
 /**
  * Builds a list from a seed value. Accepts an iterator function, which returns
- * either false to stop iteration or an array of length 2 containing the value
- * to add to the resulting list and the seed to be used in the next call to the
- * iterator function.
+ * either an empty array to stop iteration, or a singleton array containing an
+ * array of length 2, with the value that's added to the resulting list and the
+ * seed to be used in the next call of the iterator function.
  *
  * The iterator function receives one argument: *(seed)*.
  *
  * @func
  * @memberOf R
  * @since v0.10.0
  * @category List
- * @sig (a -> [b]) -> * -> [b]
+ * @sig (a -> Maybe [b]) -> * -> [b]
```

Comment:
Pushed `(b -> Array (Pair a b)) -> b -> Array a` \o/
Yep, code is a lot better than words 👍 thanks for the patience.